src/classes/STG_SettingsManager_TEST.cls

```diff
@@ -769,6 +769,424 @@ private with sharing class STG_SettingsManager_TEST {
             'Installment Auto Creation Setting should be correctly converted to respected custom label');
     }
 
+    /**
+    * @description Should prevent read only user to delete opportunity naming settings
+    * as they do not have delete permission
+    */
+    @isTest
+    private static void shouldPreventDeleteOppNamingWhenUserDoesNotHaveDeletePermission() {
+        Opportunity_Naming_Settings__c settings = getOppNameSettings();
+        insert settings;
+
+        Test.startTest();
+
+        System.runAs(buildReadOnlyUser()) {
+            Test.setCurrentPage(Page.STG_PanelOppNaming);
+            STG_PanelOppNaming_CTRL panel = new STG_PanelOppNaming_CTRL();
+            panel.currentONS = settings;
+            panel.currentONSName = settings.Name;
+
+            panel.delONS();
+
+            UTIL_UnitTestData_TEST.assertPageHasError(System.Label.stgDeletePermissionError);
+            System.assertEquals(1, panel.listOppNameSettings.size());
+            System.assertEquals(1, panel.mapOppNameSettings.size());
+            System.assert([select id, name from Opportunity_Naming_Settings__c].size() > 0);
+        }
+
+        Test.stopTest();
+
+    }
+
+    /**
+    * @description Verify if admin users can delete opportunity naming settings
+    */
+    @isTest
+    private static void shouldVerifyDeleteOppNamingWhenUserHaveAdminPermission() {
+        Opportunity_Naming_Settings__c settings = getOppNameSettings();
+        insert settings;
+
+        Test.startTest();
+
+        Test.setCurrentPage(Page.STG_PanelOppNaming);
+        STG_PanelOppNaming_CTRL panel = new STG_PanelOppNaming_CTRL();
+        system.assertEquals('idPanelOppNaming', panel.idPanel());
```

Comment:
`System`  instead `system`, but is it required to verify the panel name here (and other new tests)?